plugins/catalog-backend/src/database/DefaultProcessingDatabase.test.ts

```diff
@@ -263,22 +276,47 @@ describe('DefaultProcessingDatabase', () => {
         savedRelations = await knex<DbRelationsRow>('relations')
           .where({ originating_entity_id: id })
           .select();
-        expect(savedRelations.length).toBe(1);
-        expect(savedRelations[0]).toEqual({
-          id: savedRelations[0].id,
+
+        const expected_newly_saved_relation = {
+          id: expect.anything(),
           originating_entity_id: id,
           source_entity_ref: 'component:default/foo',
           type: 'memberOf',
           target_entity_ref: 'component:default/foo',
-        });
+        };
+
+        const expected_previous_saved_relation = {
+          id: expect.anything(),
+          originating_entity_id: expect.any(String),
+          source_entity_ref: 'component:default/foo',
+          type: 'memberOf',
+          target_entity_ref: 'component:default/foo',
+        };
+
+        const sqlite_expected_newly_saved_relation = {
+          originating_entity_id: id,
+          source_entity_ref: 'component:default/foo',
+          type: 'memberOf',
+          target_entity_ref: 'component:default/foo',
+        };
+
+        const sqlite_expected_previous_saved_relation = {
+          originating_entity_id: expect.any(String),
+          source_entity_ref: 'component:default/foo',
+          type: 'memberOf',
+          target_entity_ref: 'component:default/foo',
+        };
+
+        expect(savedRelations.length).toBe(1);
+        expect(savedRelations[0]).toEqual(
```

Comment:
it would be more readable ( in my opinion ) to remove the ternary within the toEqual callback and wrap these into separate if blocks. Just my $0.02